core/src/main/java/io/aiven/klaw/config/ManageDatabase.java

```diff
@@ -565,6 +565,19 @@ public void loadTopicsForOneTenant(int tenantId) {
     topicsPerTenant.put(tenantId, handleDbRequests.getAllTopics(tenantId));
   }
 
+  public void addTopicToCache(int tenantId, Topic topic) {
+    log.info("addTopicToCache {} {}", tenantId, topic);
+    // If the topic does not already exist in the cache simply add it.
+    if (!topicsPerTenant.get(tenantId).contains(topic)) {
+      topicsPerTenant.get(tenantId).add(topic);
+    } else {
+      // If the topic does exist in the cache remove it and then add it to get any updates or
+      // changes that may have been added wtihout getting duplication.
+      topicsPerTenant.get(tenantId).remove(topic);
```

Comment:
why to remove ? if old topic is in dev, and from sync if we want to add topic to prd, it would be a new topic object which needs to be cached I think

Reply:
So I wanted to make sure as it is a list that we didn't end up in a situation where the same object was placed in the cache twice. I also was thinking it might be possible that the original entry maybe had different, advanced settings or replication factor etc so if it is synced across we should add the new one.

Let me know if you think i am over thinking that

Reply:
Every topic has a unique topic id, and its associated to a different environment. so we don't have to remove right

Reply:
Oh interesting, I might have a separate defect here then, that i will look into. and resolve.